src/transformStateWithClones.js

```diff
@@ -6,8 +6,40 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  let copy = { ...state };
+
+  for (const object of actions) {
+    switch (object.type) {
+      case 'addProperties':
+        copy = Object.assign({}, copy, object.extraData);
+        break;
+
+      case 'removeProperties':
+        copy = removeProperties(copy, object.keysToRemove);
+        break;
+
+      case 'clear':
+        copy = {};
+        break;
+    }
+    result.push(copy);
```

Comment:
Ensure that `copy` is a new object each time it is pushed to the `result` array. This is crucial to avoid unintended mutations in subsequent iterations.